src/test/java/com/devonfw/sample/archunit/ArchitectureTest.java

```diff
@@ -15,16 +15,23 @@ public class ArchitectureTest {
 
   @ArchTest
   private static final ArchRule shouldOnlyAccessValidLayers = //
-      layeredArchitecture().consideringAllDependencies() //
-          .layer("common").definedBy("..common..") //
-          .layer("logic").definedBy("..logic..") //
-          .layer("dataaccess").definedBy("..dataaccess..") //
-          .layer("service").definedBy("..service..") //
-          .layer("batch").definedBy("..batch..") //
-          .layer("ui").definedBy("..ui..") //
-          // TODO
-          .because("Dependency of technical layers violates architecture rules.");
+      layeredArchitecture().consideringAllDependencies() //      
+      .layer("common").definedBy("..common..") //
+      .layer("logic").definedBy("..logic..") //
+      .layer("dataaccess").definedBy("..dataaccess..") //
+      .layer("service").definedBy("..service..") //
+      .layer("client").definedBy("..client..")
+      .layer("batch").definedBy("..batch..")
+
+      .whereLayer("client").mayNotBeAccessedByAnyLayer()
+      .whereLayer("batch").mayOnlyBeAccessedByLayers( "logic")
+      .whereLayer("service").mayOnlyBeAccessedByLayers("client")
+      .whereLayer("logic").mayOnlyBeAccessedByLayers("service", "batch")
+      .whereLayer("dataaccess").mayOnlyBeAccessedByLayers("logic")
+      .whereLayer("common").mayOnlyBeAccessedByLayers("common", "dataaccess", "logic", "service")
```

Comment:
Actually `common` may be accessed by any layer. Further, always code from one layer may always be allowed to call code from the same layer on the level of layer checks (`common` should not be explicitly listed here and `batch` or `client` are missing). Only for cross-component calls there is a difference (Rules C3-C7 that are not part of this story). 

So here I would just use `mayNotAccessAnyLayer()` forbidding access from `common` to other but not vice versa.

Reply:
If I use mayNotAccessAnyLayer() on the `common` layer, I encounter my primary problem:
The `common` layer depends on other packages like _java.*_ and _javax.*_ and throws errors. The mayNotAccessAnyLayer rule seems too restrictive.

I suggest adding batch and client layers to the mayOnlyBeAccessedByLayers() rule and excluding `common` itself as mentioned by you. 
With this rule no errors regarding the dependencies of `common` upon java.* and javax.* arise. 

Alternatively, I need to ignore the given dependencies by the ignoreDependency() rule if the mayNotAccessAnyLayer rule is preferred.

Reply:
Great. Lets see via tests, if that already does the job or we need to explicitly list all other layers here.